src/common/perf_counters.cc

```diff
@@ -354,9 +374,27 @@ void PerfCounters::reset()
 }
 
 void PerfCounters::dump_formatted_generic(Formatter *f, bool schema,
-    bool histograms, const std::string &counter) const
+    bool histograms, bool dump_labeled, const std::string &counter) const
 {
-  f->open_object_section(m_name.c_str());
+
+  if(dump_labeled) {
+    std::string_view perf_counter_name = ceph::perf_counters::key_name(m_name);
+    f->open_object_section(perf_counter_name);
+
+    f->open_object_section("labels");
+    for (auto label : ceph::perf_counters::key_labels(m_name)) {
+      // don't dump labels with empty label names
+      if(label.first == "") {
```

Comment:
@cbodley my latest commit changed this to only filter our labels who don't have names so `"rgw": { "labels": { "Bucket": "", "User": "user1", "": "val_x" } } "put_b": 1048576, }` will appear as `"rgw": { "labels": { "Bucket": "", "User": "user1", } "put_b": 1048576, }`

From one of your previous comments I can see why we should allow empty label values, but I can't see why we should allow empty label names. Am I missing something?

Reply:
+1.. Empty values are possible, Empty label names is an error and it does no have any senses.